BitwardenShared/Core/Platform/Services/StateService.swift

```diff
@@ -1186,6 +1193,12 @@ actor DefaultStateService: StateService { // swiftlint:disable:this type_body_le
         try getAccount(userId: userId).profile.userDecryptionOptions?.hasMasterPassword ?? true
     }
 
+    func getUserId(email: String) async -> String? {
+        guard let state = appSettingsStore.state else { return nil }
+        let account = state.accounts.values.first { $0.profile.email == email }
```

Comment:
⚠️ The user might have set up more than one account with the same email in different servers, so we should be checking all the accounts that match the email and not only the first one. Therefore I think this should be returning an array of possible user ids.

**Example**
Logged in Accounts:
- Account1:
  - Email: e1@example.com
  - UserId: 1
  - Base Url: US
- Account2:
  - Email: e2@example.com
  - UserId: 2
  - Base Url: US
- Account3:
  - Email: e1@example.com
  - UserId: 3
  - Base Url: EU
 
If we attempt to log in using `e1@example.com` in `EU` server then it should display the alert for the `Account3` but currently it won't happen because of only taking into consideration the first one.

Reply:
Ah, good catch, I updated it to handle that scenario.